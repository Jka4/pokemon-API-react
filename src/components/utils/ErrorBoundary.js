import React from 'react';

export default class ErrorBoundary extends React.PureComponent {
	state = { error: null };

	componentDidCatch(error, errorInfo = {}) {
		this.setState({ error });
		errorInfo.ErrorType = 'ReactError';
	}

	render() {
		if (this.state.error) {
			return (
				<div className='errorBoundary'>
					<div className="textErrorBoundary">Ooops...</div>
				</div>
			);
		}

		return this.props.children;
	}
}