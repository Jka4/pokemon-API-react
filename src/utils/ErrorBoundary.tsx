import React from "react";

export default class ErrorBoundary extends React.PureComponent {
  state = { error: null };

  componentDidCatch(error: any, errorInfo: any) {
    this.setState({ error });
    errorInfo.ErrorType = "ReactError";
  }

  render() {
    if (this.state.error) {
      return (
        <div className="errorBoundary">
          <div className="textErrorBoundary">Ooops...</div>
        </div>
      );
    }

    return this.props.children;
  }
}
