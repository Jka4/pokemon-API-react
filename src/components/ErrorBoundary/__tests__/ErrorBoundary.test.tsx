import * as React from 'react';

import ErrorBoundary from '../ErrorBoundary';
import { mount } from 'enzyme';

describe('<ErrorBoundary/> ', () => {
  const Something = () => null;

  it('should display an ErrorMessage if wrapped component throws', () => {
    const container = mount(
      <ErrorBoundary>
        <Something />
      </ErrorBoundary>,
    );

    const error = new Error('test');

    container.find(Something).simulateError(error);
    expect(container).toMatchSnapshot();
  });
});
