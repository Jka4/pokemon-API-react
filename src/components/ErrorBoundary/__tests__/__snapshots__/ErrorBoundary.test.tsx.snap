// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary/>  should display an ErrorMessage if wrapped component throws 1`] = `
<ErrorBoundary>
  <styled.div>
    <div
      className="sc-bdfBwQ hFpCER"
    >
      <styled.div>
        <div
          className="sc-gsTCUz gfTRSZ"
        >
          Ooops...
        </div>
      </styled.div>
    </div>
  </styled.div>
</ErrorBoundary>
`;
